.github/workflows/checks.yml

```diff
@@ -0,0 +1,49 @@
+name: Code Checks
+
+on:
+  pull_request:
+  push:
+    branches: [main]
+
+jobs:
+  merge-conflicts:
+    runs-on: ubuntu-latest
+    name: Find merge conflicts
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Check for merge conflicts
+        uses: olivernybroe/action-conflict-finder@v4.0
```

Comment:
wouldn't git do it for us anyway

Reply:
Yeah, its was in the previous pre-commit, that's why I added it.
Removing it as GitHub itself block merge if conflicts is there.